src/components/diagrams/diagram.js

```diff
@@ -0,0 +1,870 @@
+import {
+    decrementNetworkAreaDiagramDepth,
+    incrementNetworkAreaDiagramDepth,
+    resetNetworkAreaDiagramDepth,
+    setFullScreenDiagram,
+} from '../../redux/actions';
+
+import { AutoSizer } from 'react-virtualized';
+
+import { useIntl } from 'react-intl';
+
+import clsx from 'clsx';
+import { RunningStatus } from '../util/running-status';
+import AlertInvalidNode from '../util/alert-invalid-node';
+import BaseEquipmentMenu from '../menus/base-equipment-menu';
+import withEquipmentMenu from '../menus/equipment-menu';
+import withLineMenu from '../menus/line-menu';
+import { equipments } from '../network/network-equipments';
+import { useIntlRef, useSnackMessage } from '@gridsuite/commons-ui';
+import { useIsAnyNodeBuilding } from '../util/is-any-node-building-hook';
+import Alert from '@mui/material/Alert';
+import {
+    isNodeBuilt,
+    isNodeReadOnly,
+    isNodeInNotificationList,
+} from '../graph/util/model-functions';
+import {
+    NetworkAreaDiagramViewer,
+    SingleLineDiagramViewer,
+} from '@powsybl/diagram-viewer';
+import {
+    SvgType,
+    getEquipmentTypeFromFeederType,
+    useDiagram,
+    computePaperAndSvgSizesIfReady,
+    commonDiagramStyle,
+    commonSldStyle,
+    commonNadStyle,
+    MAX_HEIGHT_SUBSTATION,
+    MAX_HEIGHT_VOLTAGE_LEVEL,
+    MAX_HEIGHT_NETWORK_AREA_DIAGRAM,
+    MAX_WIDTH_SUBSTATION,
+    MAX_WIDTH_VOLTAGE_LEVEL,
+    MAX_WIDTH_NETWORK_AREA_DIAGRAM,
+    NoSvg,
+    LOADING_WIDTH,
+} from './diagram-common';
+import makeStyles from '@mui/styles/makeStyles';
+import DiagramHeader from './diagram-header';
+import DiagramFooter from './diagram-footer';
+
+const customSldStyle = (theme) => {
+    return {
+        '& .arrow': {
+            fill: theme.palette.text.primary,
+        },
+    };
+};
+
+const useStyles = makeStyles((theme) => ({
+    divSld: commonSldStyle(theme, customSldStyle(theme)),
+    divNad: commonNadStyle(theme),
+    ...commonDiagramStyle(theme),
+}));
+
+let initialWidth, initialHeight;
+
+const Diagram = forwardRef((props, ref) => {
+    const [svg, setSvg] = useState(NoSvg);
+
+    const dispatch = useDispatch();
+    const { snackError } = useSnackMessage();
+    const intlRef = useIntlRef();
+
+    const svgRef = useRef();
+    const diagramViewerRef = useRef();
+
+    const theme = useTheme();
+    const classes = useStyles();
+    const intl = useIntl();
+
+    const {
+        openDiagramView,
+        minimizeDiagramView,
+        togglePinDiagramView,
+        closeDiagramView,
+    } = useDiagram();
+
+    const network = useSelector((state) => state.network);
+
+    const currentNode = useSelector((state) => state.currentTreeNode);
+
+    const fullScreenDiagram = useSelector((state) => state.fullScreenDiagram);
+
+    const notificationIdList = useSelector((state) => state.notificationIdList);
+
+    const [forceState, updateState] = useState(false);
+
+    const networkAreaDiagramDepth = useSelector(
```

Comment:
That would be nice to separate hooks and function required for SLD or for NAD. I think it's  a bit damaging to have them in the same component and file

Reply:
Agree, but maybe not today and in this current US, as this would take (me) too much time.